### PR TITLE
Problem with solving simple separable ODE fixed

### LINKED ISSUE #22155: Problem with solving simple separable ODE
I want to solve by SymPy simple separable equation y'=e^(y-x), but SymPy can't solve it!
```
from sympy import *
x = symbols("x")
y = Function("y")
eq = Eq(Derivative(y(x), x), exp(y(x)-x))
dsolve(eq, y(x))
```
But it solves successfuly similar equation y'=e^(y+x).

Follоwing code
```
eq = Eq(Derivative(y(x), x), exp(y(x)-x))
print(classify_ode(eq,y(x)))
```
returns

`('factorable', 'separable', '1st_power_series', 'lie_group', 'separable_Integral')
`

The method for 'factorable' seems broken. Explicitly using the second method,

`dsolve(eq, y(x), hint='separable')
`

quickly returns the correct solution.

For the second equation y'=e^(y+x), the list of methods is

`('separable', '1st_power_series', 'lie_group', 'separable_Integral')
`

so in this case the problematic method 'factorable' is not used. 

Experiments have shown that those equations cannot be solved if their exponents have a minus sign.

### PR DESCRIPTION
**Pull request by mohajain**
#### Brief description of what is fixed or changed
Fixes #22155

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by mohajain** [comment]
I am getting errors in some cases after applying the diff. The code is now showing different results. Should I change the results present to the expected values?
@oscarbenjamin

**Comment by oscarbenjamin** [comment]
> Should I change the results present to the expected values?

I think so.

**Comment by mohajain** [comment]
I have fixed all the errors @oscarbenjamin

**Comment by oscarbenjamin** [comment]
Great. A new test should be added for the case that failed in the original issue. I think it was solved by the separable solver so it should go somewhere in here:
https://github.com/sympy/sympy/blob/6dd5c539952c2e8a6a611a421af373689709e726/sympy/solvers/ode/tests/test_single.py#L1645

File: sympy/solvers/tests/test_pde.py [hunk 1]
```diff
@@ -231,4 +231,4 @@ def test_pdsolve_variable_coeff():
 
     eq = exp(2*x)*(u.diff(y)) + y*u - u
     sol = pdsolve(eq, hint='1st_linear_variable_coeff')
-    assert sol == Eq(u, exp((-y**2 + 2*y + 2*F(x))*exp(-2*x)/2))
+    assert sol == Eq(f(x, y), F(x)*exp(-y*(y - 2)*exp(-2*x)/2))
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_pde.py:234]: This should still be `u` rather than `f(x, y)` to keep it consistent with the examples above

**Comment by Mohitbalwani26** [comment]
Add this example in `test_ode.py` under `test_classify_ode()`. It will help in future if we make Factorable hint as part of preprocessing.

**Comment by mohajain** [comment]
I have made the changes requested.

File: sympy/solvers/ode/tests/test_ode.py [hunk 2]
```diff
@@ -305,6 +306,13 @@ def test_classify_ode():
     a = classify_ode(2*x*f(x)*f(x).diff(x) + (1 + x)*f(x)**2 - exp(x), f(x), dict=True, hint='Bernoulli')
     assert sorted(a.keys()) == ['Bernoulli', 'Bernoulli_Integral', 'default', 'order', 'ordered_hints']
 
+    # test issue 22155
+    a = classify_ode(f(x).diff(x) - exp(f(x) - x),f(x))
```
Comment (oscarbenjamin) [inline sympy/solvers/ode/tests/test_ode.py:310]: There should be a space after a comma

Comment (mohajain) [reply sympy/solvers/ode/tests/test_ode.py]: Done!

**Comment by oscarbenjamin** [comment]
@Mohitbalwani26 does this look good to you?

**Comment by Mohitbalwani26** [comment]
Yes, It looks good. Thanks

**Pull request merged**